packages/datetime/src/timePicker.tsx

```diff
@@ -285,11 +299,17 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
                 this.setState({ millisecondText: text });
                 break;
         }
+        if (this.props.inputProps) {
```

Comment:
please avoid implicit type coercion like this, instead use `if (this.props.inputProps != null)`

Reply:
actually, how about abstracting this repeated check into a helper method?

```ts
private maybeInvokeInputEventHandler(handlerName: "onChange" | "onBlur" | "onKeyDown" | "onFocus") {
  if (this.props.inputProps != null) {
    BlueprintUtils.safeInvoke<React.SyntheticEvent<HTMLInputElement>, void>(this.props.inputProps[handlerName], e);
  }
}
```

Reply:
sure, good suggestion 👍